src/Rule.php

```diff
@@ -20,6 +20,34 @@ abstract class Rule implements RuleInterface, ParametrizedRuleInterface, Formatt
      */
     private $when = null;
 
+    /**
+     * To create an instance of a rule class, the static factory method `rule()` must be implemented
+     * in the final class.
+     *
+     * Simple example:
+     *
+     * ```php
+     * public static function rule(): self
+     * {
+     *     return new self();
+     * }
+     * ```
+     *
+     * Example with parameters:
+     *
+     * ```php
+     * public static function rule(int $count): self
+     * {
+     *     $rule = new self();
+     *     $rule->count = $count;
+     *     return $rule;
+     * }
+     * ```
+     */
+    final protected function __construct()
```

Comment:
Should it be private?

Reply:
No, in child classes should be available constructor. For example, see rule `AtLeast`.